primitives/arithmetic/src/per_things.rs

```diff
@@ -67,7 +67,8 @@ pub trait PerThing:
 		let b = Self::ACCURACY;
 		// if Self::ACCURACY % 100 > 0 then we need the correction for accuracy
 		let c = rational_mul_correction::<Self::Inner, Self>(b, a, 100.into(), Rounding::Nearest);
-		Self::from_parts(a / 100.into() * b + c)
+		let base: Self::Inner = 100.into();
+		Self::from_parts((a / base).saturating_mul(b).saturating_add(c))
```

Comment:
Fuck, I'm too dumb xD How does this work?

When `a` is `50u32`. And I do `a / base`, I will get `0`?

Reply:
yes you are right, this function doesn't work, and should have tests generated by macro

Reply:
I wasn't sure how this function works either tbh :D I will have a look.

Reply:
Why not just use the `Self::from_rational_approximation` here rather than a custom impl? This function is quite old and well tested as well.

Reply:
Addressed, I just used the well tested `from_rational`.